packages/gatsby/src/internal-plugins/query-runner/page-query-runner.js

```diff
@@ -74,7 +74,12 @@ exports.runQueuedActions = runQueuedActions
 // Wait until all plugins have finished running (e.g. various
 // transformer plugins) before running queries so we don't
 // query things in a 1/2 finished state.
-emitter.on(`API_RUNNING_QUEUE_EMPTY`, runQueuedActions)
+emitter.on(`API_RUNNING_QUEUE_EMPTY`, async api => {
```

Comment:
Is this reliable? Does calling `createNode` always finish with an `onCreateNode` API call? E.g. in the future we might be encouraging people to create pages in response to nodes being created in which case the last API call would be `onCreatePage`.

Reply:
That's good question. I'll try creating pages in `onCreateNode` and report back.